### DISCUSSION TITLE
Export Metrics+Logging via quarkus-opentelemetry

### DISCUSSION THREAD

**Discussion by abhiaagarwal** [keycloak/keycloak#35012] [Ideas]
Hi there,

I'm utilizing Keycloak in an environment where I have to collect logs, metrics, and tracing data for security reasons. Keycloak already seems to support metrics and logging handlers in a non-opentel way, but since the broader ecosystem seems to be moving towards opentel, it might be a good idea as to plug in with opentel-connector-contrib and ship logs/metrics directly without any intermediary services.

Quarkus already seems to have (experimental) support for this, I'm going to try and see if I can build keycloak with those options during the work week.  https://quarkus.io/guides/opentelemetry#opentelemetry-metrics-guide

**Comment by francesc-jorda** [comment]
Hi, we're about to try the same in our environment. 
Did you achive to send logs and metrics?

**Reply by abhiaagarwal** [reply]
I ended up using the rsyslog receiver and scraping prometheus metrics manually, as setting the quarkus variables did nothing. Would definitely like to give it a shot again!

**Comment by mabartos** [comment]
Hi guys! There's a new initiative to leverage OTel for Logs and Metrics.

See https://github.com/keycloak/keycloak/issues/41261
